### ISSUE TITLE
Wires.reset does not perform as expected for marshals with fields of Bytes

### ISSUE THREAD

**Issue by RobAustin** [OpenHFT/Chronicle-Wire#45]
see net.openhft.chronicle.wire.WiresTest#test

```
 public static class Base {
        Bytes base = Bytes.elasticByteBuffer();
    }

    @Test
    public void test() {

        Base base1 = new Base();
        base1.base.clear().append("value1");

        Base base2 = new Base();
        base2.base.clear().append("value2");

        Wires.reset(base1);
        Wires.reset(base2);

        base1.base.clear().append("value1");

        Assert.assertEquals("", base2.base.toString());

    }
```

**Comment by RobAustin** [comment]
added example

**Comment by epickrram** [comment]
I've committed a workaround for now to get the build passing.

https://github.com/OpenHFT/Chronicle-Wire/commit/bb3473578b68e830b06852525fd16d3f4267caff

Unclear as to what the long-term solution should be, as the need to add an annotation is non-obvious, and garbage is generated on each reset() operation.

**Comment by peter-lawrey** [comment]
I agree. We should be able to do this eithout an annotation or creating
garbage on each message.

On 15 Jun. 2017 12:32, "Mark Price" <notifications@github.com> wrote:

I've committed a workaround for now to get the build passing.

bb34735
<https://github.com/OpenHFT/Chronicle-Wire/commit/bb3473578b68e830b06852525fd16d3f4267caff>

Unclear as to what the long-term solution should be, as the need to add an
annotation is non-obvious, and garbage is generated on each reset()
operation.

—
You are receiving this because you were assigned.

Reply to this email directly, view it on GitHub
<https://github.com/OpenHFT/Chronicle-Wire/issues/45#issuecomment-308693615>,
or mute the thread

**Comment by epickrram** [comment]
Fixed for `Bytes` and `StringBuilder` - the cleanest solution to this may be documentation + allowing the user to register a custom `FieldAccess` handler in `WireMarshaller`.

**Comment by RobAustin** [comment]
@epickrram could you document this as you suggest and close this issue.

**Comment by RobAustin** [comment]
@epickrram could you pick this up today ?

**Comment by epickrram** [comment]
Will do...

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.101](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
